copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/WorkerSinkTask.java

```diff
@@ -207,18 +213,6 @@ public WorkerConfig workerConfig() {
         log.debug("Task {} subscribing to topics {}", id, topics);
         newConsumer.subscribe(Arrays.asList(topics), new HandleRebalance());
 
-        // Seek to any user-provided offsets. This is useful if offsets are tracked in the downstream system (e.g., to
-        // enable exactly once delivery to that system).
-        //
-        // To do this correctly, we need to first make sure we have been assigned partitions, which poll() will guarantee.
-        // We ask for offsets after this poll to make sure any offsets committed before the rebalance are picked up correctly.
-        newConsumer.poll(0);
```

Comment:
Ah, sorry, I forgot to add a note about that. It actually is. At this point in the code it's actually impossible for the Task to have changed the offsets -- the `createConsumer()` call happens before the task has been started. So this code was effectively a nop. With this updated version you create the consumer, create task context, join the group to ensure you have partition assignment, initialize + start the task (during which they could potentially request offsets to change). Then, upon your first poll, the first thing we check is if there are any user-submitted offsets to rewind to (which we need to check every time we get control back from the SinkTask). So the correct rewinding behavior falls out naturally without any special handling in the initialization case.